### ISSUE TITLE
Never ending loop cursor for contacts and some other resources

### ISSUE THREAD

**Issue by clintonskakun** [agilecrm/rest-api#25]
I've debugged and double checked this for a long amount of time. Making sure my code wasn't causing the issue and working very hard to make everything work perfectly on my end.

This is what I've experienced:

Looping over to get contacts using the cursor is running in an endless loop (causing my scipts to run for 24+ hours or forever). It never appears to reach the end of the list. Alternatively I'd like to have an option just to pass a page number instead of a cursor, feel like that would work better.

I check for same ids, I check for null cursor (which seems to be in every request).

I can't get this to work on my end. The horrible part is sometimes it works and other times not. The randomness makes it hard to test and completely unreliable.

If you guys can help me out, and help out the developers community, I'd appreciate it a million times over. I want to keep my AgileCRM clients happy and I really appreciate the effort on your part as well.

**Comment by graut** [comment]
Hi @clintonskakun ,

It should work properly with cursor.

I am attaching two videos which will explain you about get list of contacts using Rest API.

About get list of contacts using Rest API
https://www.youtube.com/watch?v=2Yui9HJXAOE

Example to  get list of contacts using Rest API
https://www.youtube.com/watch?v=jJAAZKqybQ4

I hope it will help you. Please let us know in case need further information.

Thanks

**Comment by clintonskakun** [comment]
@graut Thank you so much for the time you took to make the videos!
I see a few things here I never saw in the docs. But thanks for laying it out.
I'm going to give this another try. If I can't get it working like you showed above, I'll create a video of my own to show you to see if you can point out anything from my code.

Thanks again!

**Comment by clintonskakun** [comment]
Ok, I just found the problem. It was my error. Can't believe it was so obvious. I've spent a looong time trying to debug this. Turned out the problem was that the cursor code or the params for curl needed to be inside quotes like "https://.../?cursor=xyz&page_size=20" in order to work. Fixing that one thing totally fixed it!!!

Thanks for your help. This at least pushed m past the last part and I learned about global_sort_key.
